source/h1_encoder.c

```diff
@@ -103,6 +103,10 @@ static void s_write_headers(struct aws_byte_buf *dst, const struct aws_http_mess
     AWS_ASSERT(wrote_all);
 }
 
+static bool s_check_info_response_status_code(int code_val) {
+    return code_val >= 100 && code_val < 200 && code_val != 101;
```

Comment:
trivial: use `AWS_HTTP_HEADER_101_SWITCHING_PROTOCOLS` and add a note that we should probably handle it the same as everything else